### ISSUE TITLE
start training router

### ISSUE THREAD

**Issue by JennyHuang19** [JennyHuang19/counterfactual_feedback#4]

**Comment by JennyHuang19** [comment]
would i be able to use the justifications to guide training? this seems like useful information!
maybe have the router respond with (<bool> , justification)?

**Comment by JennyHuang19** [comment]
would be able to pull additional data from the turns that don't have negative feedback (but try wildchat4m first).

**Comment by JennyHuang19** [comment]
filtered out sensitive code-related data.
filter_policy_violations.py

**Comment by JennyHuang19** [comment]
Job Details:

Training Method: Supervised
Suffix: router-train
Base model: gpt-4.1-nano-2025-04-14

**Comment by JennyHuang19** [comment]
final train accuracy: 0.955
final validation accuracy: 0.892

training set size: 406
validation set size: 352

model for ground truth label creating: gpt-5-nano
router model: gpt-4.1-nano-2025-04-14

**Comment by JennyHuang19** [comment]
training v2. train_filtered.jsonl is a larger dataset.

train set size: 1,463 turns.
validation set size: 352

**Comment by JennyHuang19** [comment]
================================================================================
EVALUATION RESULTS
================================================================================

Total test samples: 250

Class distribution:
  'true' samples: 125
  'false' samples: 125

Overall Performance:
  Accuracy:  0.7320 (183/250)

Metrics for 'true' class (requires reasoning):
  Precision: 0.7231
  Recall:    0.7520
  F1 Score:  0.7373

Confusion Matrix:
Predicted False  Predicted True
Actual False                89              36
Actual True                 31              94

================================================================================

**Comment by JennyHuang19** [comment]
performance roughly the same between base models
- gpt-4.1-nano-2025-04-14
- gpt-4o-mini

**Comment by JennyHuang19** [comment]
base model choice: around the same performance between

-  gpt-4.1-nano-2025-04-14
- gpt-4o-mini
